pkg/controllers/federate/controller.go

```diff
@@ -541,10 +550,19 @@ func (c *FederateController) handleExistingFederatedObject(
 	logger := klog.FromContext(ctx)
 
 	logger.V(3).Info("Checking if federated object needs update")
-	needsUpdate, err := updateFederatedObjectForSourceObject(ftc, sourceObject, fedObject)
+	needsUpdate, sourceObjNeedsUpdate, err := updateFederatedObjectForSourceObject(ctx, ftc, sourceObject, fedObject, c.eventRecorder)
 	if err != nil {
 		return false, fmt.Errorf("failed to check if federated object needs update: %w", err)
 	}
+
+	if sourceObjNeedsUpdate {
```

Comment:
If I understood correctly, with this PR, the new annotations are processed as follows:
1. User writes `kubeadmiral.io/migration-info` on the source object
2. federate controller does the following when converting the source object to the federated object:
	2.1. parses it and validates it
    2.2. converts it to `kubeadmiral.io/custom-migration-info` and writes it to the fedobject
    2.3. converts it to `kubeadmiral.io/latest-migration-info` and writes it to the source object
3. scheduler reads `kubeadmiral.io/custom-migration-info` and schedules accordingly

Migration/Eviction should be in the scheduler's domain, and letting federate controller process it breaks abstraction and doesn't really fit well into the current architecture. Without clear boundaries, controllers become hard to maintain and understand.

Meanwhile, step 2.3 doesn't synchronize with step 3, so the reported `latest-migration-info` may not even have taken effect.

I propose the following solution to the above problems:
1. We already have the mechanism to copy source annotations to the fed object (see `classifyAnnotations`). We can use it to copy `kubeadmiral.io/migration-info` to the fedObject.
2. Leave the validation and processing to the scheduler. Once the scheduler has rescheduled based on the new migration config, it writes `kubeadmiral.io/latest-migration-info` to the fedObject at the same time it updates the fedObject placement.
3. `sourcefeedback` in federate controller copies `kubeadmiral.io/latest-migration-info` to the source object.

cc @limhawjia @mrlihanbo Please share your opinions too.

Reply:
It is a little stranger to put the `sourcefeedback` in federate controller